manifests/init.pp

```diff
@@ -346,89 +346,89 @@
-  $read_conn_lifetime                      = $puppetdb::params::read_conn_lifetime,
-  $confdir                                 = $puppetdb::params::confdir,
-  $vardir                                  = $puppetdb::params::vardir,
-  $manage_firewall                         = $puppetdb::params::manage_firewall,
-  $java_args                               = $puppetdb::params::java_args,
-  $merge_default_java_args                 = $puppetdb::params::merge_default_java_args,
-  $max_threads                             = $puppetdb::params::max_threads,
-  $command_threads                         = $puppetdb::params::command_threads,
-  $concurrent_writes                       = $puppetdb::params::concurrent_writes,
-  $store_usage                             = $puppetdb::params::store_usage,
-  $temp_usage                              = $puppetdb::params::temp_usage,
-  $disable_update_checking                 = $puppetdb::params::disable_update_checking,
-  $certificate_whitelist_file              = $puppetdb::params::certificate_whitelist_file,
-  $certificate_whitelist                   = $puppetdb::params::certificate_whitelist,
-  $database_max_pool_size                  = $puppetdb::params::database_max_pool_size,
-  $read_database_max_pool_size             = $puppetdb::params::read_database_max_pool_size,
-  Boolean $automatic_dlo_cleanup           = $puppetdb::params::automatic_dlo_cleanup,
-  String[1] $cleanup_timer_interval        = $puppetdb::params::cleanup_timer_interval,
-  Integer[1] $dlo_max_age                  = $puppetdb::params::dlo_max_age,
-  Postgresql::Pg_password_encryption $postgresql_password_encryption = $puppetdb::params::password_encryption,
-  Optional[Stdlib::Absolutepath] $java_bin = $puppetdb::params::java_bin,
+  Stdlib::Host                                                         $listen_address                    = $puppetdb::params::listen_address,
+  Variant[Stdlib::Port::User, Pattern[/\A[0-9]+\Z/]]                   $listen_port                       = $puppetdb::params::listen_port,
+  Boolean                                                              $disable_cleartext                 = $puppetdb::params::disable_cleartext,
+  Boolean                                                              $open_listen_port                  = $puppetdb::params::open_listen_port,
+  Stdlib::Host                                                         $ssl_listen_address                = $puppetdb::params::ssl_listen_address,
+  Variant[Stdlib::Port::User, Pattern[/\A[0-9]+\Z/]]                   $ssl_listen_port                   = $puppetdb::params::ssl_listen_port,
+  Boolean                                                              $disable_ssl                       = $puppetdb::params::disable_ssl,
+  Optional[Boolean]                                                    $open_ssl_listen_port              = $puppetdb::params::open_ssl_listen_port,
```

Comment:
I think instead of `Optional[Boolean]`, the value in params.pp should be changed to `false` and then use `Boolean` as type.

Reply:
agreed